addons/account_sequence/models/account_move.py

```diff
@@ -0,0 +1,19 @@
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    _sql_constraints = [(
+        'unique_name', "", "Another entry with the same name already exists.",
+    )]
+
+    def _auto_init(self):
+        super()._auto_init()
+        self.env.cr.execute("""
+            CREATE UNIQUE INDEX IF NOT EXISTS account_move_unique_name
+            ON account_move(name, journal_id) WHERE (state = 'posted' AND name != '/');
+        """)
```

Comment:
Prefer `odoo.tools.sql.create_[unique_]index`, IIRC `create_index if not exists` will first take a write-exclusive lock on the table *then* check if the index already exists, which can have negative effects when the index already exists, because it will prevent concurrent updates to the table for the entire rest of the transaction.

Reply:
Concurrent updates in `_auto_init`?

Reply:
I don't understand.
Are you talking about the `IF NOT EXISTS` part? That this will lock the table even if the index already exists?

Reply:
Ok, should be good now then